### DISCUSSION TITLE
Packaging script with executable

### DISCUSSION THREAD

**Discussion by kasajian** [babashka/babashka#1418] [Q&A]
It'd be cool if you can build this and include (embed) custom script(s) into the executable that runs on default.  That way you can create single-file executables that don't require babahska to be delivered separately.   Has anyone done this, and written down the steps you went through to do it?

**Comment by borkdude** [comment]
Hey @kasajian, welcome! We use Github Discussions for Q & A over here.

You can make a self-contained executable using caxa: https://github.com/babashka/babashka/wiki/Self-contained-executable

**Comment by kasajian** [comment]
Well that's awesome.  Hello world at 134 MB.  I assume this is normal and I'm not doing anything wrong.  bb itself is 74 MB.

**Reply by borkdude** [reply]
@kasajian A common pitfall is that you will package the package itself inside the package if you have packaged multiple times. Make sure you start with a clean situation when packaging. See the last note at the wiki page.

**Comment by holyjak** [comment]
There is now https://github.com/babashka/babashka/wiki/Self-contained-executable#uberjar
